cmake/CMakeLists.txt

```diff
@@ -581,7 +581,7 @@ endif()
 
 set_target_properties(lammps PROPERTIES OUTPUT_NAME lammps${LAMMPS_MACHINE})
 set_target_properties(lammps PROPERTIES SOVERSION ${SOVERSION})
-target_include_directories(lammps PUBLIC $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
+target_include_directories(lammps PUBLIC $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/lammps>)
```

Comment:
@akohlmey I don't care much about the rest of the PR, but getting this include path fix into the next stable release would be nice.

Reply:
What is exactly that this change going to fix?

Reply:
If you are trying to build `COUPLE/simple` against an installed lammps, the include directory coming for `lammps::lammps` is currently wrong.
`${CMAKE_INSTALL_INCLUDEDIR}` has no `library.h` in it, but `${CMAKE_INSTALL_INCLUDEDIR}/lammps` has it, hence the include path in the install interface needs to be the latter.

This change is needed as we changed the example back from `#include "lammps/library.h"` to `#include "library.h`.

This change is mainly about consistency.

The rest of the PR is mainly to avoid something like #2384 by build the example as part of the tests.

Reply:
I'll do this one change because it (obviously) doesn't do any harm. However, as I have stated repeatedly, the "make install" functionality is not something that is at this point in time something that has been properly reviewed and is not something intended to work. For this to work properly, first some more changes to the LAMMPS C++ headers is required and a different folder tree. The option is there because nobody really paid much attention to it while other, more relevant stuff was added.
I have been been tempted multiple times to propose removing it because it causes me to spend time on reviewing and discussing changes related to it. For now the intended way to use the library interface is the traditional one, i.e. to use it "in place" from the source and build folders.